src/core/config/merge.js

```diff
@@ -34,6 +35,13 @@ const merge = (target, ...sources) => {
     merged.domNode = domNode
   }
 
+  Object.entries(merged).forEach(([key, value]) => {
```

Comment:
I put this change here instead of in `query` because of `urls.primaryName`. 

The issue is that if we set `urls.primaryName`, we’ll have 
```js
config: { urls: { primaryName: 'string' } }
```
 so when using deep extend where in config of lower priority we'll have:
```js
config: { urls: [{ url: 'url1', name: 'string' }] }
```
The result of this will just be the object:
```js
config: { urls: { primaryName: 'string' } }
```

I tried setting it by doing something like
```js
queryOptions.urls = []
set(queryOptions, key, value)
```

Which seemed fine, as we had an empty array with `primaryName` property, but then when merging it with the other options, I would just get an empty array. 
A quick test for this:
```js
  const foo = { urls: [] }
  set(foo, "urls.primaryName", "string")
  const bar = { urls: ["a", "b"] }
  const result = deepExtend({}, bar, foo))
```

Another question is if we even want to read `urls.primaryName` here. If we want to change it so that we only allow these types of options to be defined in the URL, we might move this to query because these lines are where we get this setting (in `master`):
https://github.com/swagger-api/swagger-ui/blob/cd7fb2778dc47ad81723ecc7013cf5aefe3405e6/src/standalone/plugins/top-bar/components/TopBar.jsx#L57-L58

https://github.com/swagger-api/swagger-ui/blob/cd7fb2778dc47ad81723ecc7013cf5aefe3405e6/src/standalone/plugins/top-bar/components/TopBar.jsx#L90

Basically: we always get the setting from `parseSearch`, so if we only want to allow this option in query, setting it here seems unnecessary.